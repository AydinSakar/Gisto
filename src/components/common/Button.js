import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseAppColor, lightText } from 'constants/colors';
import Icon from 'components/common/Icon';

const ButtonComponent = styled.div`
  background: ${baseAppColor};
  border: 1px solid ${baseAppColor};
  color: ${lightText};
  border-radius: 3px;
  height: 30px;
  margin: 10px 0;
  font-weight: 200;
  font-size: 14px;
  display: flex;
  align-items: center;
  width: 50%;
  
  &[invert] {
    background: #fff;
    border: 1px solid #fff;
    color: ${baseAppColor};
  }
`;

const Button = ({ icon, children }) => (
  <ButtonComponent>
    <Icon invert type={ icon }/> { children }
  </ButtonComponent>
);

Button.propTypes = {
  icon: PropTypes.string,
  children: PropTypes.node
};

export default Button;
