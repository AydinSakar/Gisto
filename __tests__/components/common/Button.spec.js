import Button from 'components/common/Button';

const propSetup = (props) => ({
  icon: 'menu',
  children: <div/>,
  width: '200px',
  height: '30px',
  invert: true,
  className: 'btn',
  ...props
});

const setup = (props) => (shallow(<Button { ...props }>link</Button>));

describe('COMPONENTS - <Button>', () => {
  test('render Button', () => {
    const wrapper = setup(propSetup());

    expect(wrapper).toMatchSnapshot();
  });

  test('handle click', () => {
    const props = {
      onClick: jest.fn(),
      invert: false
    };
    const wrapper = setup(propSetup(props));

    wrapper.simulate('click');

    expect(props.onClick).toHaveBeenCalled();
  });
});
