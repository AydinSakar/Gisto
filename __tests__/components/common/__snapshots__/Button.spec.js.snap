// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS - <Button> render Button 1`] = `
<Styled(button)
  className="btn"
  height="30px"
  invert={true}
  title="link"
  width="200px"
>
  <Styled(span)
    color="#3F84A8"
    type="menu"
  />
   
  link
</Styled(button)>
`;
